DSharpPlus.Commands/CommandsExtension.cs

```diff
@@ -192,8 +192,8 @@ public void Setup(DiscordClient client)
     public void AddCommand(Delegate commandDelegate) => this.commandBuilders.Add(CommandBuilder.From(commandDelegate));
     public void AddCommand(Type type, params ulong[] guildIds) => this.commandBuilders.Add(CommandBuilder.From(type, guildIds));
     public void AddCommand(Type type) => this.commandBuilders.Add(CommandBuilder.From(type));
-    public void AddCommands(Assembly assembly, params ulong[] guildIds) => AddCommands(assembly.GetTypes(), guildIds);
-    public void AddCommands(Assembly assembly) => AddCommands(assembly.GetTypes());
+    public void AddCommands(Assembly assembly, params ulong[] guildIds) => AddCommands(assembly.GetTypes().Where(type => !type.IsNested), guildIds);
+    public void AddCommands(Assembly assembly) => AddCommands(assembly.GetTypes().Where(type => !type.IsNested));

```

Comment:
this introduces an undocumented and breaking restriction that classes cannot contain top-level command classes.

Reply:
Like this?

```cs
public class Commands
{
    [Command("ping")]
    public class PingCommand
    {
        [Command("pong")]
        public static ValueTask PongAsync(CommandContext context) => context.RespondAsync("Pong!");
    }

    [Command("hello")]
    public class HelloCommand
    {
        [Command("world")]
        public static ValueTask WorldAsync(CommandContext context) => context.RespondAsync("Hello, World!");
    }
}
```